### PR TITLE
Make CI work with numpy v1.24

### LINKED ISSUE #99: NumPy 1.24 lets XPASS ufunc tests
The test suite fails with NumPy 1.24 because 5 ufunc tests unexpectedly XPASS:

```

[   16s] + pytest-3.8 --ignore=_build.python38 --ignore=_build.python39 --ignore=_build.python310 -v --pyargs erfa -k 'not test_version_with_embedded_liberfa'
[   17s] ============================= test session starts ==============================
[   17s] platform linux -- Python 3.8.16, pytest-7.1.2, pluggy-1.0.0 -- /usr/bin/python3.8
[   17s] cachedir: .pytest_cache
[   17s] rootdir: /home/abuild/rpmbuild/BUILD/pyerfa-2.0.0.1, configfile: setup.cfg
[   17s] plugins: doctestplus-0.12.0
[   17s] collecting ... collected 279 items / 1 deselected / 278 selected
[   17s] 
...
[   17s] =================================== FAILURES ===================================
[   17s] __________________________________ test_bi00 ___________________________________
[   17s] [XPASS(strict)] do not yet support no-input ufuncs
[   17s] _________________________________ test_fk5hip __________________________________
[   17s] [XPASS(strict)] do not yet support no-input ufuncs
[   17s] ___________________________________ test_ir ____________________________________
[   17s] [XPASS(strict)] do not yet support no-input ufuncs
[   17s] ___________________________________ test_zp ____________________________________
[   17s] [XPASS(strict)] do not yet support no-input ufuncs
[   17s] ___________________________________ test_zr ____________________________________
[   17s] [XPASS(strict)] do not yet support no-input ufuncs
[   17s] =========================== short test summary info ============================
[   17s] FAILED tests/test_ufunc.py::test_bi00
[   17s] FAILED tests/test_ufunc.py::test_fk5hip
[   17s] FAILED tests/test_ufunc.py::test_ir
[   17s] FAILED tests/test_ufunc.py::test_zp
[   17s] FAILED tests/test_ufunc.py::test_zr
[   17s] ============ 5 failed, 272 passed, 1 skipped, 1 deselected in 0.83s ============
```

### PR DESCRIPTION
**Pull request by avalentino**
Closes #99

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -48,7 +48,7 @@ doctest_optionflags =
     IGNORE_EXCEPTION_DETAIL
 text_file_format = rst
 addopts = --doctest-rst
-xfail_strict = true
+# xfail_strict = true  # behavior change in numpy v1.24
```
Comment (mhvk) [inline setup.cfg:51]: Ideally, we make the failure numpy-version dependent instead of removing this protection.

File: tox.ini [hunk 2]
```diff
@@ -36,7 +39,7 @@ description =
 # The following provides some specific pinnings for key packages
 deps =
     oldestdeps: numpy==1.17.*
-    devdeps: :NIGHTLY:numpy
```
Comment (mhvk) [inline tox.ini:39]: We should also remove the `indexserver` entry on line 11.

File: tox.ini [hunk 3]
```diff
@@ -56,7 +56,7 @@ description = invoke sphinx-build to build the HTML docs
 extras = docs
 commands =
     pip freeze
-    sphinx-build -W -b html . _build/html
```
Comment (mhvk) [inline tox.ini:56]: Why this change? Ideally, we fix whatever causes the warnings!

Was it this warning from the log that caused the problem?
```
/home/docs/checkouts/readthedocs.org/user_builds/pyerfa/envs/100/lib/python3.7/site-packages/sphinx_astropy/conf/v1.py:227: UserWarning: matplotlib's plot_directive could not be imported. Inline plots will not be included in the output
```
This sounds like a bug in `sphinx-astropy`, though if takes setting `matplotlib` as a docs dependency to remove the warning and get the docs to build with `-W`, then perhaps that is worth it. Pinging @astrofrog...

**Review by mhvk — COMMENTED** [review]
Thanks! Looks mostly good, but I wonder a bit about removing some protections against future regressions.

File: setup.cfg (hunk 1 above)
Comment (avalentino) [reply setup.cfg]: done

File: tox.ini (hunk 2 above)
Comment (avalentino) [reply tox.ini:39]: done

File: tox.ini (hunk 3 above)
Comment (avalentino) [reply tox.ini]: It seems that this is no longer an issue ... even if it is not clear to me why

File: erfa_generator.py [hunk 4]
```diff
@@ -607,7 +607,7 @@ def xfail(self):
                 # Works on newer numpy
                 return "np.__version__ < '1.21', reason='needs numpy >= 1.21'"
             else:
-                return "reason='do not yet support no-input ufuncs'"
+                return "np.__version__ < '1.24', reason='do not yet support no-input ufuncs'"
```
Comment (mhvk) [inline erfa_generator.py:610]: Great! Could you also change the docstring?

File: erfa/tests/helper.py [hunk 5]
```diff
@@ -80,7 +80,7 @@ def treat_deprecations_as_exceptions():
     except ImportError:
         pass
     else:
-        if LooseVersion(matplotlib.__version__) < '3':
```
Comment (mhvk) [inline erfa/tests/helper.py:83]: I wonder if we should just go for `matplotlib.__version__[0] < '3'` and avoid using `packaging` at all?

Or, indeed, shall we just remove this? 3.0 appeared in 2018!

**Review by mhvk — COMMENTED** [review]
One little further request and one suggestion...

File: tox.ini (hunk 3 above)
Comment (mhvk) [reply tox.ini]: ??? Well, we'll take it!

File: erfa/tests/helper.py (hunk 5 above)
Comment (avalentino) [reply erfa/tests/helper.py:83]: OK I went for the first option, just to stay on the safe side ...

File: erfa_generator.py (hunk 4 above)
Comment (avalentino) [reply erfa_generator.py]: assume you mean the `reason` ... done.

File: erfa_generator.py [hunk 6]
```diff
@@ -607,7 +607,7 @@ def xfail(self):
                 # Works on newer numpy
                 return "np.__version__ < '1.21', reason='needs numpy >= 1.21'"
             else:
-                return "np.__version__ < '1.24', reason='do not yet support no-input ufuncs'"
+                return "np.__version__ < '1.24', reason='numpy < 1.24 do not support no-input ufuncs'"
```
Comment (mhvk) [inline erfa_generator.py:610]: Hmm, I actually meant the docstring of `xfail`, sorry for not being clearer!

**Review by mhvk — APPROVED** [review]
Totally trivial last bit - just because I wasn't clear enough earlier.

Will approve now, since really this doesn't matter!

File: erfa/tests/helper.py [hunk 7]
```diff
@@ -6,7 +6,7 @@
 import sys
 import types
 import warnings
-from distutils.version import LooseVersion
+import packaging.version
```
Comment (mhvk) [inline erfa/tests/helper.py:9]: I thought this was no longer needed now?

Comment (avalentino) [reply erfa/tests/helper.py:9]: yep, sorry

**Review by mhvk — APPROVED** [review]
Looks all OK now. Fine to squash the commits in the merge?

**Comment by avalentino** [comment]
> Looks all OK now. Fine to squash the commits in the merge?

yes, sure

File: tox.ini [hunk 8]
```diff
@@ -35,8 +36,8 @@ description =
 
 # The following provides some specific pinnings for key packages
 deps =
-    oldestdeps: numpy==1.17.*
-    devdeps: :NIGHTLY:numpy
+    oldestdeps: numpy==1.20.*
```
[the remark this answers is not in the document]
Comment (pllim) [reply tox.ini:39]: FWIW, I pinned the oldest deps in astropy LTS branch to pyerfa 2.0.0.1, so this isn't end of the world for astropy but it is concerning that this happened in a bug fix release.

[the remark this answers is not in the document]
Comment (pllim) [reply tox.ini:39]: Also, not particularly related to this PR but might be related to the 2.0.0.2 release also, astropy 32-bit job started to time out around the same time of the release but only in backport branches. I might have to also pin pyerfa for those... not sure yet. I am rerunning those jobs again to confirm that it is consistently timing out, which will take a while.
